packages/generator/src/edmx-to-vdm/common/complex-type.ts

```diff
@@ -88,16 +98,21 @@ export function transformComplexTypesBase(
           jsType: typeMapping.jsType,
           fieldType: typeMapping.fieldType,
           isComplex,
+          isEnum,
           isCollection
         };
-      }).filter(filterUnknownPropertyTypes)
+      })
     };
   });
 }
 
 export function getTypeMappingComplexProperties(
   typeName: string,
-  formattedTypes: MapType<any>
+  enumTypes: VdmEnumType[],
```

Comment:
Minor & genral understanding: It seems like for complex type there is a rule between the typeName and the actual type. See `complexTypeFieldType(formattedTypes[withoutPrefix]`. For enums it also seems like this:  `e.originalName === getPostfix(name)` if you look in the `findEnumType`. For me both approaches are fine. Using a list of types and find the right one or extract from the typeName. However, we should not mix.

Reply:
This function is only used only once, where we iterate all the properties of a complex type.
Without enum, the current master checks the following condition like a `switch` keyword:
- isEdm
- isCollection
- isComplex
- error if not match (enum is filtered before)
Now enum comes, I removed the filter and added a new condition so the `switch` it becomes:
- isEdm
- isCollection
- isComplex
- isEnum
- error if not match